drivers/serial/uart_nrfx_uart.c

```diff
@@ -216,6 +238,62 @@ static int uart_nrfx_err_check(struct device *dev)
 	return error;
 }
 
+static int uart_nrfx_configure(struct device *dev,
+			       const struct uart_config *cfg)
+{
+	nrf_uart_parity_t parity;
+	nrf_uart_hwfc_t hwfc;
+
+	if (cfg->stop_bits != UART_CFG_STOP_BITS_1) {
+		return -ENOTSUP;
+	}
+
+	if (cfg->data_bits != UART_CFG_DATA_BITS_8) {
+		return -ENOTSUP;
+	}
+
+	switch (cfg->flow_ctrl) {
+	case UART_CFG_FLOW_CTRL_NONE:
+		hwfc = NRF_UART_HWFC_DISABLED;
+		break;
+	case UART_CFG_FLOW_CTRL_RTS_CTS:
+		if (get_dev_config(dev)->rts_cts_pins_set) {
+			hwfc = NRF_UART_HWFC_ENABLED;
+		} else {
+			return -ENOTSUP;
+		}
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	switch (cfg->parity) {
+	case UART_CFG_PARITY_NONE:
+		parity = NRF_UART_PARITY_EXCLUDED;
+		break;
+	case UART_CFG_PARITY_EVEN:
+		parity = NRF_UART_PARITY_INCLUDED;
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	if (baudrate_set(dev, cfg->baudrate) != 0) {
+		return -ENOTSUP;
```

Comment:
I think that value `-EINVAL` is better here.

Reply:
API specifies that -ENOTSUP will be returned if configuration is not supported by driver. Same applies to other places